draft-ietf-quic-recovery.md

```diff
@@ -398,8 +417,80 @@ A packet sent on an RTO alarm MUST NOT be blocked by the sender's congestion
 controller. A sender MUST however count these bytes as additional bytes in
 flight, since this packet adds network load without establishing packet loss.
 
+## Multiple Packet Number Space Optimizations {#optimizations}
+
+There are cases where one may be able to gain recovery information from
+acknowledgements of packets in another packet number space, but they rely
+on complex assumptions about the peer’s processing and acknowledgement
+algorithms.  Even those are unable to quickly recover from cases such as
+losing the client's Initial, but receiving the 0-RTT packets.  Below are
+three different optimizations in increasing complexity that minimize
+handshake latency.
+
+### Empty ACK frames
+
+The EMPTY_ACK frame enables faster recovery of lost Initial and Handshake
+packets when the only other outstanding packets are undecryptable.
```

Comment:
I think we can improve on this for Handshake packets. Let's consider the case where in its last flight the client sends a large client cert and the CFIN, and then starts sending 1-RTT data. When receiving the 1-RTT packets before processing the CFIN, the server would send empty ACKs. However, it would make more sense to send a non-empty ACK to inform the client exactly which packets are missing.
Maybe the better solution is to introduce a new frame for this purpose?

Reply:
What would it send non-empty acks of?  It can't even decode the PN of the 1RTT packets.